common/v2/services/Store/Account/AccountProvider.tsx

```diff
@@ -86,14 +87,55 @@ export const AccountProvider: React.FC = ({ children }) => {
         state.updateAccount(existingAccount.uuid, existingAccount);
       }
     },
-    updateAccountsBalances: toUpdate => {
+    updateAllAccountsAssets: async (storeAccounts, assets) =>
+      Promise.all(
+        storeAccounts.map(async (storeAccount) => {
+          // Find all tokens with a positive balance for given account, and add those tokens to the assets array of the account
+          const assetBalances = await getAllTokensBalancesOfAccount(storeAccount, assets);
+          const positiveAssetBalances = Object.entries(assetBalances).filter(
+            ([_, value]) => !value.isZero()
+          );
+
+          const existingAccount = accounts.find((x) => x.uuid === storeAccount.uuid);
+
+          if (existingAccount) {
+            const newAssets: AssetBalanceObject[] = positiveAssetBalances.reduce(
+              (
+                tempAssets: AssetBalanceObject[],
+                [contractAddress, balance]: [string, BigNumber]
+              ) => {
+                const tempAsset = assets.find((x) => x.contractAddress === contractAddress);
+                if (tempAsset) {
+                  tempAssets.push({
+                    uuid: tempAsset.uuid,
+                    balance: balance.toString(10),
```

Comment:
In memory it's better for the balance to be stored as a BigNumber.
It's only when it's saved to localstorage that we need to coerce to string.
And that's the responsibility of demarshalling

Reply:
This is how it had been implemented in AccountProvider all-along